tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -28,7 +28,44 @@ use tracing_core::{
 
 /// A `Layer` which filters spans and events based on a set of filter
 /// directives.
-// TODO(eliza): document filter directive syntax?
+///
+/// # Directives
+///
+/// A filter consists of one or more directives. Directives match [`Span`]s and [`Event`]s
+/// and specify a maximum verbosity [`level`] to enable for those that match. The directive
+/// syntax is similar to the one presented in `env_logger`. The syntax consists
+/// of four parts `target[span{field=value}]=level`.
+///
+/// - `target` matches on the target that the event comes from, generally this will be the
+/// module path. Examples, `h2`, `tokio::net`, etc. For more info checkout the [`Metadata`] docs.
+/// - `span` matches on the span name that you want to filter on. If this is supplied with a `target`
+/// it will match on all filter spans within that `target.
```

Comment:
"match on all filter spans" doesn't really make a lot of sense, can we revise this?